### PR TITLE
Cleanup in C# generated code

### PR DESCRIPTION
**Pull request by jtattermusch**
Removes the "serviceName" argument from ServerServiceDefinition.CreateBuilder. (api breakage, but only in generated code, so regenerating fixes the problem).
Fixes  https://github.com/grpc/grpc/issues/6581.

Removes deprecated interface for client side and server-side stubs.
Fixes https://github.com/grpc/grpc/issues/6582.

### CODE REVIEW

File: src/csharp/Grpc.Core/ServerServiceDefinition.cs [hunk 1]
```diff
@@ -63,28 +63,24 @@ private ServerServiceDefinition(Dictionary<string, IServerCallHandler> callHandl
...
         {
-            readonly string serviceName;
             readonly Dictionary<string, IServerCallHandler> callHandlers = new Dictionary<string, IServerCallHandler>();
 
             /// <summary>
             /// Creates a new instance of builder.
             /// </summary>
-            /// <param name="serviceName">The service name.</param>
-            public Builder(string serviceName)
+            public Builder()
```
Comment (kpayson64) [inline src/csharp/Grpc.Core/ServerServiceDefinition.cs]: nit: Usually empty constructors are defined in one line.

**Comment by kpayson64** [comment]
Given that this is a breaking API change, can you update examples/csharp as well?

**Comment by jtattermusch** [comment]
I marked this PR as "doc update needed". We usually update the examples once the change is actually released, because otherwise the example would be broken for people until the release happens.

File: src/csharp/Grpc.Core/ServerServiceDefinition.cs (hunk 1 above)
Comment (jtattermusch) [inline src/csharp/Grpc.Core/ServerServiceDefinition.cs]: I plan to introduce an automatic code formatter at some point, and after that I'll fix all of these at once.

**Comment by kpayson64** [comment]
lgtm

**Comment by jtattermusch** [comment]
green here: https://grpc-testing.appspot.com/job/gRPC_pull_requests/9577/

**Pull request merged**